examples/1d/elixir_euler_nonperiodic.jl

```diff
@@ -7,15 +7,16 @@ using Trixi
 
 equations = CompressibleEulerEquations1D(1.4)
 
-initial_conditions = initial_conditions_convergence_test
+initial_condition = initial_condition_convergence_test
 
 source_terms = source_terms_convergence_test
 
 # you can either use a single function to impose the BCs weakly in all
 # 1*ndims == 2 directions or you can pass a tuple containing BCs for
 # each direction
-# boundary_conditions = boundary_conditions_convergence_test
-boundary_conditions = ntuple(n->boundary_conditions_convergence_test, 2)
+# boundary_condition = boundary_condition_convergence_test
+boundary_condition = (x_neg=boundary_condition_convergence_test,
+                      x_pos=boundary_condition_convergence_test)
```

Comment:
I wasn't sure how to handle this part: We allow passing both a single function/item as BC or multiple ones. Hence, we do not strictly know whether we will have one or multiple functions/items. On the other hand, we will always have multiple boundaries...

Reply:
Internally, boundary conditions is a vector-like container, so there's always a multiple of them. The single-function option is just for laziness/convenience, so I wouldn't consider it as relevant. To me it makes sense to call the individual boundary conditions `boundary_condition_x`, but the collection of them should be plural, since - as you pointed out - there are always multiple boundaries.

Reply:
Fixed in 2d1fc09.